packages/website/bin/page-generator/get-docs-info.js

```diff
@@ -4,26 +4,34 @@
 const path = require('path');
 const fs = require('fs-extra');
 
-const processDirectory = dirPath =>
-  fs
-    .readdirSync(dirPath)
-    .map(fname => ({
-      id: path.parse(fname).name,
-      fullPath: path.resolve(dirPath, fname),
-    }))
-    .map(({ id, fullPath }) => {
-      if (fs.statSync(fullPath).isFile()) {
+const processDirectory = dirPath => {
+  try {
+    return fs
+      .readdirSync(dirPath)
+      .map(fname => ({
+        id: path.parse(fname).name,
+        fullPath: path.resolve(dirPath, fname),
+      }))
+      .map(({ id, fullPath }) => {
+        if (fs.statSync(fullPath).isFile()) {
+          return {
+            id,
+            path: fullPath,
+          };
+        }
+
         return {
           id,
-          path: fullPath,
+          children: processDirectory(fullPath),
         };
-      }
-
-      return {
-        id,
-        children: processDirectory(fullPath),
-      };
-    });
+      });
+  } catch (err) {
```

Comment:
Instead of catching errors, could we use [fs.existsSync()](https://nodejs.org/api/fs.html#fs_fs_existssync_path) to identify when files don't exist?

I think the complexity here is that the first call to processDirectory needs to behave differently to subsequent calls, right?

What we really care about is whether the original dirPath exists, so we can stop once we've done that.

So we could do:

```
if (!fs.existsSync(dirPath)) return null;

return fs
  .readdirSync(dirPath
...
```

We probably need to be a little more defensive so:

```
if (!fs.existsSync(dirPath) || !fs.statSync(dirPath).isDirectory()) return null;
```